task/generate-labels/0.1/generate-labels.yaml

```diff
@@ -0,0 +1,73 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "konflux"
+  name: generate-labels
+spec:
+  description: |
+    Generate labels based on templates.
+
+    Usage may look like the following.
+
+    > - name: generate-labels
+    >   params:
+    >   - name: LABEL_TEMPLATES
+    >     value: '["release=$SOURCE_DATE_EPOCH", "build-date=$SOURCE_DATE"]'
+
+    The following environment variables are defined for use in LABEL_TEMPLATES
+
+    * ACTUAL_DATE - a date time string containing the time this task runs, formatted +'%Y-%m-%dT%H:%M:%SZ'
+    * ACTUAL_DATE_EPOCH - the timestamp at the time this task runs
+    * SOURCE_DATE - a date time string containing the provided source timestamp, formatted +'%Y-%m-%dT%H:%M:%SZ'
+    * SOURCE_DATE_EPOCH - the timestamp provided as a param meant to represent the timestamp at which the source was last modified
+
+  params:
+  - name: label-templates
+    description: A json array of templates that should be rendered and exposed as a list of labels
+    type: string
+  - name: source-date-epoch
+    description: A standardised environment variable for build tools to consume in order to produce reproducible output.
```

Comment:
Is this the date epoch for the sources used or just the source of the epoch to use for calculations?

Reply:
Both?

It is the source of the epoch to use for calculations.

It _can_ be the date epoch for the sources used and _should_ be, but nothing requires that it.

Reply:
You are saying that this date should be the same as the date of the git commit? Would this pattern only really make sense in a reproducible build scenario (and/or a hermetic build) as we wouldn't be pulling in dynamic content to differentiate from the content committed to git?